ethcore/src/block.rs

```diff
@@ -266,8 +267,9 @@ impl<'x> OpenBlock<'x> {
 		r.block.base.header.set_extra_data(extra_data);
 		r.block.base.header.note_dirty();
 
-		let gas_floor_target = ::std::cmp::max(gas_range_target.0, engine.params().min_gas_limit);
-		engine.populate_from_parent(&mut r.block.base.header, parent, gas_floor_target, gas_range_target.1);
+		let gas_floor_target = cmp::max(gas_range_target.0, engine.params().min_gas_limit);
+		let gas_ceil_target = cmp::max(gas_range_target.1, gas_floor_target);
```

Comment:
looks fine to me except for the possible case where `ceil` < `floor`... at this point I'm not sure it's safe to assume anything :)